migrations/1694797181616_brc20-counts-by-address-event.ts

```diff
@@ -0,0 +1,69 @@
+/* eslint-disable @typescript-eslint/naming-convention */
+import { MigrationBuilder, ColumnDefinitions } from 'node-pg-migrate';
+
+export const shorthands: ColumnDefinitions | undefined = undefined;
+
+export function up(pgm: MigrationBuilder): void {
+  pgm.createTable('brc20_counts_by_address_event_type', {
+    address: {
+      type: 'text',
+      notNull: true,
+      primaryKey: true,
+    },
+    deploy: {
+      type: 'bigint',
+      notNull: true,
+      default: 0,
+    },
+    mint: {
+      type: 'bigint',
+      notNull: true,
+      default: 0,
+    },
+    transfer: {
+      type: 'bigint',
+      notNull: true,
+      default: 0,
+    },
+    transfer_send: {
+      type: 'bigint',
+      notNull: true,
+      default: 0,
+    },
+  });
+  pgm.sql(`
+    INSERT INTO brc20_counts_by_address_event_type (address, deploy) (
+      SELECT address, COUNT(*) AS deploy FROM brc20_deploys GROUP BY address
+    ) ON CONFLICT (address) DO UPDATE SET deploy = EXCLUDED.deploy
+  `);
+  pgm.sql(`
+    INSERT INTO brc20_counts_by_address_event_type (address, mint) (
+      SELECT address, COUNT(*) AS mint FROM brc20_mints GROUP BY address
+    ) ON CONFLICT (address) DO UPDATE SET mint = EXCLUDED.mint
+  `);
+  pgm.sql(`
+    INSERT INTO brc20_counts_by_address_event_type (address, transfer) (
+      SELECT from_address AS address, COUNT(*) AS transfer FROM brc20_transfers GROUP BY from_address
+    ) ON CONFLICT (address) DO UPDATE SET transfer = EXCLUDED.transfer
+  `);
+  pgm.sql(`
+    INSERT INTO brc20_counts_by_address_event_type (address, transfer_send) (
+      SELECT from_address AS address, COUNT(*) AS transfer_send
+      FROM brc20_transfers
+      WHERE to_address IS NOT NULL
+      GROUP BY from_address
+    ) ON CONFLICT (address) DO UPDATE SET transfer_send = EXCLUDED.transfer_send
+  `);
+  pgm.sql(`
+    INSERT INTO brc20_counts_by_address_event_type (address, transfer_send) (
+      SELECT to_address AS address, COUNT(*) AS transfer_send
+      FROM brc20_transfers
+      WHERE to_address <> from_address
+      GROUP BY to_address
+    ) ON CONFLICT (address) DO UPDATE SET transfer_send = EXCLUDED.transfer_send
+  `);
```

Comment:
Not sure I follow these last migration.
.transfer_send is overridden.

3. we count the SENT transfers for the from_address (number of transfers sent)
4. we count the SENT transfers for the to_address (number of transfers received, unrelated) -- it would write over the previous from_address count?


> also: does this also have a GROUP BY row of NULL address? since NULL <> from_address? wouldn't insert since address it NOTNULL, but could this error in cases?

Reply:
Ah, good catch, this should increase the count instead of replacing it. I'll make this change